server/genserver.go

```diff
@@ -460,6 +481,24 @@ func (h handler) {{.Op}}Handler(ctx context.Context, w http.ResponseWriter, r *h
 		return
 	}
 
+	{{if .HasPaging}}
+	if !swag.IsZero(nextPageID) {
+		{{if .SingleStringPathParameter}}
+		{{.InputVarName}} = nextPageID
```

Comment:
Yeah, since this is in the `SingleStringPathParameter` branch (and single-string-path-parameter only happens for _required_ single-string parameter functions, making it not a pointer).